custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js

```diff
@@ -83,6 +83,19 @@ function MonthFilterController($scope, $location, $uibModal, storageService, dat
     var vm = this;
 
     // used by mobile dashboard
+    var reportStartDates = {
+        'sdd': new Date(2019, 1),
+    };
+
+    var isSDD =  $location.path().indexOf('service_delivery_dashboard') !== -1;
+
+    vm.startYear = 2017;
+    vm.startMonth = 3;
+
+    if (isSDD) {
+        vm.startYear = reportStartDates['sdd'].getFullYear();
+        vm.startMonth = reportStartDates['sdd'].getMonth() + 1; //adding 1 due to zero based indexing
+    }
```

Comment:
it concerns me that this logic now lives in multiple places because if we add new reports with different start dates we'll have to update things in a few places and could easily forget one. A few mitigating strategies:

1. Extract this logic to one or more shared services or utility functions somewhere. 
2. (if 1 is too much work) extract `reportStartDates` to a constant that is used everywhere so at least that part is maintained separately.

What do you think?

Reply:
I agree. I will make changes